library/core/src/iter/traits/iterator.rs

```diff
@@ -1535,34 +1535,32 @@ pub trait Iterator {
     ///
     /// # Examples
     ///
-    /// Basic usage:
+    /// This demonstrates a use case that needs `by_ref`:
```

Comment:
"Basic usage:" is a standard we use across the docs; if you grep for this you'll see a *lot*. Consistency matters; I would prefer to leave this alone.

Reply:
I find it confusing here, because it's referring to an example that doesn't use `by_ref` that should.

Reply:
Ah you're right. So, this is my bad, then. What we *should* do is have the up-front "basic usage" example of something simple that *does* work, and then show the one that doesn't afterwards.

Reply:
Just to be clear, are you suggesting adding an additional example that we would put before the current one?

Reply:
Yes, the standard style is to have a very very basic example as the very first one, then more complicated stuff afterwards.